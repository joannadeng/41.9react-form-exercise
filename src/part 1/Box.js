import React from "react";
import "./Box.css"

const Box = ({removeBox, id, color, width, height}) => {
    
    return (
        <div className="Box" style={{backgroundColor:color,width:width, height:height}}>
            <button onClick={() => removeBox(id)}>X</button>
        </div>
        
    )
}

export default Box;