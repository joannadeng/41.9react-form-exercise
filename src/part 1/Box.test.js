import {render} from '@testing-library/react';
import Box from './Box';

it ('it should render without crashing', () => {
    render(<Box />)
})

it ("should match snapshot", () => {
    const {asFragment} = render(<Box id={1} color={"red"} width={30} height={40}/>);
    expect(asFragment()).toMatchSnapshot()
})

