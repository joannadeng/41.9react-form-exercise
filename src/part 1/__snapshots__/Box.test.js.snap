// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    style="background-color: red; width: 30px; height: 40px;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
